wear/src/main/java/io/homeassistant/companion/android/tiles/ThermostatTile.kt

```diff
@@ -0,0 +1,325 @@
+
+                    tile.setTileTimeline(
+                        primaryLayoutTimeline(
+                            this@ThermostatTile,
+                            requestParams,
+                            null,
+                            R.string.tile_fetch_entity_error,
+                            R.string.refresh,
+                            ActionBuilders.LoadAction.Builder().build()
+                        )
+                    ).build()
+                }
+            }
+        }
+    }
+
+    override fun onTileResourcesRequest(requestParams: ResourcesRequest): ListenableFuture<Resources> = serviceScope.future {
+        Resources.Builder()
+            .setVersion(requestParams.version)
+            .addIdToImageMapping(
+                RESOURCE_REFRESH,
+                ResourceBuilders.ImageResource.Builder()
+                    .setAndroidResourceByResId(
+                        ResourceBuilders.AndroidImageResourceByResId.Builder()
+                            .setResourceId(io.homeassistant.companion.android.R.drawable.ic_refresh)
+                            .build()
+                    ).build()
+            )
+            .build()
+    }
+
+    override fun onTileAddEvent(requestParams: EventBuilders.TileAddEvent) = runBlocking {
+        withContext(Dispatchers.IO) {
+            val dao = AppDatabase.getInstance(this@ThermostatTile).thermostatTileDao()
+            if (dao.get(requestParams.tileId) == null) {
+                dao.add(ThermostatTile(id = requestParams.tileId))
+            } // else already existing, don't overwrite existing tile data
+        }
+    }
+
+    override fun onTileRemoveEvent(requestParams: EventBuilders.TileRemoveEvent) = runBlocking {
+        withContext(Dispatchers.IO) {
+            AppDatabase.getInstance(this@ThermostatTile)
+                .thermostatTileDao()
+                .delete(requestParams.tileId)
+        }
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        serviceScope.cancel()
+    }
+
+    private suspend fun timeline(tileConfig: ThermostatTile, entity: Entity<Map<String, Any>>, targetTemperature: Float?, temperatureUnit: String): Timeline =
+        Timeline.fromLayoutElement(
+            LayoutElementBuilders.Box.Builder().apply {
+                val currentTemperature = entity.attributes["current_temperature"]
+                var hvacAction = entity.attributes["hvac_action"].toString()
+                val state = entity.state
+                val hvacActionColor = when (hvacAction) {
+                    "heating" -> getColor(R.color.colorDeviceControlsThermostatHeat)
+                    "cooling" -> getColor(R.color.colorDeviceControlsDefaultOn)
+                    else -> 0x00000000
+                }
+
+                hvacAction = when (state) {
+                    "off" -> "off"
+                    "unavailable" -> "off"
+                    else -> hvacAction
+                }
```

Comment:
Changing `hvacAction` to `off` when `unavailable`, also makes it _look_ like unavailable climate entities are off (see: `friendlyHvacAction`), that seems unwanted to me? Instead you should just edit line 259/264 to also disable the buttons when `unavailable`. (The code suggestion assumes you'll edit those lines.)

Try to avoid a `var` if you immediately reassign it.

```suggestion
                val hvacAction = entity.attributes["hvac_action"].toString()
                val hvacActionColor = when (hvacAction) {
                    "heating" -> getColor(R.color.colorDeviceControlsThermostatHeat)
                    "cooling" -> getColor(R.color.colorDeviceControlsDefaultOn)
                    else -> 0x00000000
                }
```

Reply:
[3d0e3e9](https://github.com/home-assistant/android/pull/4959/commits/3d0e3e9c7be365e03d6f5c4b6f80333c1f51c386)